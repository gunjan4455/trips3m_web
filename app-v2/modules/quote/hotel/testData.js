export const hotelInfo = {
  quoteCityId: 35268058,
  hotelId: 3436,
  tripDayId: 38054617,
  nights: '2',
  city: 'Calangute',
  name: 'Sunkissed Plaza',
  rating: 3,
  address: 'Calangute -  North Goa, Gauravaddo Calangute\r\nBardez GOA 403516 INDIA',
  taUrl: 'http://www.tripadvisor.com/Hotel_Review-g306995-d3916311-Reviews-Sunkissed_Plaza-Calangute_Goa.html',
  dId: 3916311,
  gId: 306995,
  roomType: ' Deluxe ',
  isAlternate: 0,
  comments: '',
  selectionStatus: null,
  description: 'Sunkissed Plaza, Goa, is set in a pleasing location near the blue waters of Calangute Beach. Situated far from the hustle-bustle of the city centre, this hotel offers a stay that is indeed tranquil. Guests can enjoy the beauty of the surroundings from their rooms with the gamut of amenities during the stay. \r\n\r\n<b>Location:</b>\r\nSunkissed Plaza is located at Gauravaddo Calangute Bardez. The Calangute Beach (Approx. 1km), and St Anthony\'s Chapel, Goa (Approx. 1km) are the nearby places of tourist interest from Sunkissed Plaza. Heavily fringed with coconut trees, Calangute Beach offers a spectacular view of the sea. Ideal for numerous water sports and beach fun, Calangute Beach, is a popular tourist destination near Sunkissed Plaza. Ancestral Goa Museum, Carambolim Lake, and Arvalam Waterfalls are the other places that are worth-exploring in Goa.\r\n\r\nDistance from Goa International Airport: 39 km (approx.)\r\nDistance from Thivim Railway Station: 22 km (approx.)\r\n\r\n<b>Hotel Features:</b>\r\nSunkissed Plaza is a ravishing estate surrounded by lush green trees. Simplistic yet comfy, Sunkissed Plaza features well-appointed rooms with all the necessary amenities. Basic amenities offered by Sunkissed Plaza are internet, 24-hour front desk, 24-hour room service, travel desk, 24-hour security and doctor on call. Business services and conference suites are available at Sunkissed Plaza for corporate travellers. There is a swimming pool for a refreshing stay. Wedding and banquet services can be arranged by Sunkissed Plaza if the patrons wish to organise social events. Game lovers can step inside the game room and spend their leisure time. For ultimate rejuvenation and relaxation, patrons can avail of Jacuzzi. Patrons can also sit back and relax at the lounge during the stay. The multi-cuisine restaurant serves tempting meals that gratify the hunger cravings of the guests.\r\n\r\n<b>Rooms:</b>\r\nStandard, premium room and deluxe rooms are available for lodging and boarding. The rooms are cosy with a modern feel to assure an undisturbed stay at Sunkissed Plaza. Elegantly appointed, these room offer a beautiful view of the lush greenery and swimming pool. Air conditioning, colour television, minibar, refrigerator, safe and telephone are the room amenities provided at Sunkissed Plaza.',
  hotelImages: [
    {
      id: 412558,
      url: 'https://res.cloudinary.com/dma3zc8eo/image/fetch/f_auto,q_auto:eco,w_630/https://dknfdeb8v69ms.cloudfront.net/attachments/pictures/412558/original/5296187_21_z.jpg'
    },
    {
      id: 412570,
      url: 'https://res.cloudinary.com/dma3zc8eo/image/fetch/f_auto,q_auto:eco,w_630/https://dknfdeb8v69ms.cloudfront.net/attachments/pictures/412570/original/5296187_22_z.jpg'
    },
    {
      id: 412585,
      url: 'https://res.cloudinary.com/dma3zc8eo/image/fetch/f_auto,q_auto:eco,w_630/https://dknfdeb8v69ms.cloudfront.net/attachments/pictures/412585/original/5296187_31_z.jpg'
    },
    {
      id: 412594,
      url: 'https://res.cloudinary.com/dma3zc8eo/image/fetch/f_auto,q_auto:eco,w_630/https://dknfdeb8v69ms.cloudfront.net/attachments/pictures/412594/original/5296187_35_z.jpg'
    },
    {
      id: 412604,
      url: 'https://res.cloudinary.com/dma3zc8eo/image/fetch/f_auto,q_auto:eco,w_630/https://dknfdeb8v69ms.cloudfront.net/attachments/pictures/412604/original/5296187_36_z.jpg'
    },
    {
      id: 412617,
      url: 'https://res.cloudinary.com/dma3zc8eo/image/fetch/f_auto,q_auto:eco,w_630/https://dknfdeb8v69ms.cloudfront.net/attachments/pictures/412617/original/5296187_26_z.jpg'
    },
    {
      id: 412627,
      url: 'https://res.cloudinary.com/dma3zc8eo/image/fetch/f_auto,q_auto:eco,w_630/https://dknfdeb8v69ms.cloudfront.net/attachments/pictures/412627/original/5296187_32_z.jpg'
    },
    {
      id: 412639,
      url: 'https://res.cloudinary.com/dma3zc8eo/image/fetch/f_auto,q_auto:eco,w_630/https://dknfdeb8v69ms.cloudfront.net/attachments/pictures/412639/original/5296187_34_z.jpg'
    },
    {
      id: 412650,
      url: 'https://res.cloudinary.com/dma3zc8eo/image/fetch/f_auto,q_auto:eco,w_630/https://dknfdeb8v69ms.cloudfront.net/attachments/pictures/412650/original/5296187_20_z.jpg'
    },
    {
      id: 412663,
      url: 'https://res.cloudinary.com/dma3zc8eo/image/fetch/f_auto,q_auto:eco,w_630/https://dknfdeb8v69ms.cloudfront.net/attachments/pictures/412663/original/5296187_28_z.jpg'
    }
  ]
};

export const hotel = {
  days: [
    2
  ],
  hotelInfos: [
    hotelInfo
  ]
};

export const hotels = [
  hotel
];
