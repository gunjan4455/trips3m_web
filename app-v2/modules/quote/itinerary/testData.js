export const itineraryInfo = [
  {
    "0": {
      order: '1',
      title: 'Private Transfer from Airport to Hotel',
      description:
        'A representative will meet and take you to you to your hotel.',
      eventTime: '15:35 - India Standard Time - 2 hours',
      sectionType: 'cab',
      imagesUrl: [
        'https://dailystormer.name/wp-content/uploads/2013/07/London-taxi.jpg',
        'https://dailystormer.name/wp-content/uploads/2013/07/London-taxi.jpg'
      ]
    },
    "1": {
      order: '4',
      title: 'Arrival at Dubai Airport.',
      description:
        'Arrival at Dubai Airport at Evening, meet and greet at Airport and we transfer you toyour hotel. Check in to hotel.',
      eventTime: 'Monday, 24 August, 2019',
      sectionType: 'itinerary',
      imagesUrl: []
    },
    "2": {
      order: '2',
      title: 'hotel details',
      description: 'hotel detail',
      eventTime: 'hotel timing',
      sectionType: 'hotel',
      imagesUrl: [
        'https://dailystormer.name/wp-content/uploads/2013/07/London-taxi.jpg'
      ],
      otherDetails: {
        name: 'hotel',
        rating: 5,
        email: '',
        phone: '44445454',
        web_url: '',
        category: '',
        hotelDesc: 'hotel description',
        tripAdvisorId: '',
        lat: '28.489743',
        long: '77.074602'
      }
    },
    "3": {
      order: '3',
      title: 'Night Safari',
      description:
        "At night, you will be taken for a memorable Night Safari experience. Being the first wildlife night park, it shows you nocturnal animalsâ€™ behavior in their natural habitat. This open-air zoo is known to house nearly 130 species of animals. After an exceptional experience, return to your hotel and stay overnight. Tip: Don't feed animals at the Night Safari",
      eventTime: '15:35 - India Standard Time',
      sectionType: 'attraction',
      imagesUrl: []
    },

    "4": {
      order: '4',
      title: 'Arrival',
      description:
        'Arrival at Dubai Airport at Evening, meet and greet at Airport and we transfer you toyour hotel. Chk in to hotel.',
      eventTime: '15:35 - India Standard Time',
      sectionType: 'flight',
      otherDetails: {
        name: 'prashant',
        image_url:
          'https://dailystormer.name/wp-content/uploads/2013/07/London-taxi.jpg',
        address: 'abc',
        rating: 4.3,
        phone: '33434',
        web_url: '',
        airport_des: 'airport description',
        lat: '28.489743',
        long: '77.074602'
      }
    },
    "5": {
      order: '3',
      title: 'Back to Hotel',
      description: 'going back to hotel after sight seeing',
      eventTime: '15:35 - India Standard Time',
      sectionType: 'attraction',
      imagesUrl: []
    }
  }
];
