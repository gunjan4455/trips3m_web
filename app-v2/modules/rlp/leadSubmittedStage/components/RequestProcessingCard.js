import React, { Component } from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import Sensor from 'react-visibility-sensor';

import { TriangleIcon, ProgressIcon, ProcessedIcon, EnvelopeIcon } from 'helpers/Icon/Icon';
import { ORGANISM_VIEWED } from 'actions/segmentEvents';

const GIconDiv = styled.div({
  width: '55px',
  lineHeight: '0px'
});

const GIconBigDiv = styled.div({
  width: '120px',
  lineHeight: '0px'
});

const GContactDiv = styled.div({
  '& a': {
    border: '1px solid #2196f3',
    padding: '10px'
  },
});


class RequestProcessingCard extends Component {
  constructor(props) {
    super(props);
    this.isEventTracked = false;
  }

  onSensorChange = (isVisible) => {
    if (isVisible && !this.isEventTracked) {
      this.isEventTracked = true;
      let section = '';
      if (this.props.message.icon === 'INACTIVE_USER') {
        section = 'Confirmation link section';
      }
      else if (this.props.message.icon === 'INCOMPLETE_INFO') {
        section = 'Edit preferences section';
      }
      else if (this.props.message.icon === 'REACTIVATE_TRIP') {
        section = 'Undo cancel section';
      }
      else if (this.props.message.icon === 'DUPLICATE_TRIP') {
        section = 'Duplicate lead created section';
      }
      this.props.trackSegment(ORGANISM_VIEWED, section, '', '');
    }
  };

  render() {
    const icon = {
      DUPLICATE_TRIP: <TriangleIcon />,
      REACTIVATE_TRIP: <TriangleIcon />,
      INACTIVE_USER: <EnvelopeIcon />,
      INCOMPLETE_INFO: <ProcessedIcon />,
    };
    const { message } = this.props;

    return (
      <Sensor onChange={this.onSensorChange}
              partialVisibility='bottom'
              delayedCall={true}>
        <div className='sbcw mb8 pl15 pr15 pb24 pt24 at_requestProcessingCard'>
          {
            message.icon === 'DEFAULT' ?
              <GIconBigDiv className='mlauto mrauto pb24'><ProgressIcon /></GIconBigDiv> :
              <GIconDiv className='mlauto mrauto pb24'> {icon[message.icon]}</GIconDiv>
          }
          {
            message.header.length ?
              <p className='fw7 sfc6 f16 text-center pb15 at_messageHeading'>{message.header}</p> :
              null
          }
          <p className='m0 f14p text-center sfc6 at_messageTxt'>{message.message}</p>
          {
            message.ctas.length ?
              <GContactDiv className='flex mt64 justifyCenter'>
                {message.ctas}
              </GContactDiv> :
              null
          }
        </div>
      </Sensor>
    );
  }
}

RequestProcessingCard.propTypes = {
  message: PropTypes.object,
  trackSegment: PropTypes.func
};

export default RequestProcessingCard;
