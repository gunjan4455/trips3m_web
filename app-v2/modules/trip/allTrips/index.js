import React from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router';
import { goBack } from 'react-router-redux';

import { reactivateBooking } from 'modules/trip/actions/tripActivities';
import AllTrips from './AllTrips';

const container = withRouter(connect(
  state => {
    return {
      trips: state.trip.list,
    };
  },
  {
    goBack,
    reactivateBooking
  })(AllTrips));

export default {
  container
};
