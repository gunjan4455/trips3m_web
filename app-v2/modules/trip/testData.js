export const tripsData = {
  loaded: true,
  loading: false,
  allIds: [
    '4809460',
    '4982568',
    '4982590',
    '4982978'
  ],
  byId: {
    '4809460': {
      id: 4809460,
      tripDays: 5,
      toLoc: 'Sri Lanka',
      firstDestinationName: 'Sri Lanka',
      startDate: '2019-07-04',
      quotesCount: 3,
      bookingStatusLabel: 'Incomplete',
      exclusivePreferencesValues: {
        cabs: 'Included',
        flights: 'Included',
        hotels: '4 Star'
      },
      budget: '38000..41000',
      fromLoc: 'Chennai, Tamil Nadu, India',
      info: '',
      adult: 2,
      status: 'Cancelled',
      child: 0,
      tripStage: 3,
      currency: 'Rupee',
      phoneNo: '+919480946000'
    },
    '4982568': {
      id: 4982568,
      tripDays: 9,
      toLoc: 'Singapore',
      firstDestinationName: 'Singapore',
      startDate: '2019-08-11',
      quotesCount: 0,
      bookingStatusLabel: 'Incomplete',
      exclusivePreferencesValues: {
        cabs: 'Included',
        flights: 'Included',
        hotels: '3 Star, 2 Star'
      },
      budget: null,
      fromLoc: 'Chennai, Tamil Nadu, India',
      info: '',
      adult: 2,
      status: 'Dormant',
      child: 0,
      tripStage: 2,
      currency: null,
      phoneNo: '+919498256800'
    },
    '4982590': {
      id: 4982590,
      tripDays: 7,
      toLoc: 'Singapore',
      firstDestinationName: 'Singapore',
      startDate: '2019-08-19',
      quotesCount: 0,
      bookingStatusLabel: 'Incomplete',
      exclusivePreferencesValues: {
        cabs: 'Included',
        flights: 'Excluded',
        hotels: '3 Star'
      },
      budget: '33500..33500',
      fromLoc: 'Chennai',
      info: '',
      adult: 4,
      status: 'Dormant',
      child: 0,
      tripStage: 3,
      currency: 'Rupee',
      phoneNo: '+919498259000'
    },
    '4982978': {
      id: 4982978,
      tripDays: 7,
      toLoc: 'Singapore',
      firstDestinationName: 'Singapore',
      startDate: '2019-08-19',
      quotesCount: 3,
      bookingStatusLabel: 'Active',
      exclusivePreferencesValues: {
        cabs: 'Included',
        flights: 'Excluded',
        hotels: '3 Star'
      },
      budget: '33500..33500',
      fromLoc: 'Chennai',
      info: '',
      adult: 4,
      status: 'Active',
      child: 0,
      tripStage: 3,
      currency: 'Rupee',
      phoneNo: '+919498297800'
    }
  }
};