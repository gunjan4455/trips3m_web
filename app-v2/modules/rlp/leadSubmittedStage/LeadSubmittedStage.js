import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

import { rescue, capitalizeEveryFirstLetter } from 'helpers/utils';
import { TT_EMAIL, TT_PHONE } from 'constants/application';
import { ORGANISM_CLICKED } from 'actions/segmentEvents';
import RequestProcessingCard from './components/RequestProcessingCard';
import Modal from 'modules/shared/GenericModal';
import CancelRequest from 'modules/trip/cancelRequest';
import { INFO_ROSTER } from 'constants/rosterTypes';

const DUPLICATE_TRIP_MSG = 'Looks like you have already put a request for this destination';

class LeadSubmittedStage extends Component {
  constructor(props) {
    super(props);

    this.state = {
      closeModal: true
    };
  }

  UNSAFE_componentWillReceiveProps(nextProps) {
    if (!this.props.activationMailSent && nextProps.activationMailSent) {
      this.props.showRoster({ type: INFO_ROSTER, message: nextProps.activationMailSuccessMsg });
    }
  }

  closePopup = () => {
    this.setState({ closeModal: true });
  };

  callbackHandler = (type) => {
    const { currentTripId, reactivateBooking, sendActivationMail } = this.props;
    switch (type) {
      case 'Reactivate_Booking':
        reactivateBooking({ tripId: currentTripId });
        break;
      case 'Resend_Confirmation':
        sendActivationMail(currentTripId);
    }
  };

  getIcon = (link) => {
    switch (link.type) {
      case 'reactivate_requested_trip':
        return 'REACTIVATE_TRIP';
      case 'send_activation_mail':
        return 'INACTIVE_USER';
      case 'edit_requested_trip':
        return 'INCOMPLETE_INFO';
      default:
        return 'DEFAULT';
    }
  };

  getCta = (link) => {
    const { currentTripId } = this.props;

    switch (link.type) {
      case 'reactivate_requested_trip':
        return <button className='btn-filled-pri-large wfull'
                       onClick={() => this.callbackHandler('Reactivate_Booking')}>Undo
          Cancel</button>;
      case 'cancel_requested_trip':
        return (<Modal fullView={true} hasCustomClose={true}
                       trigger={<span className='f14 fw7 pfc3' onClick={this.trackSegmentEvents}>Cancel Request</span>}
                       customCloseTrigger={this.state.closeModal}>
          <div className='sbcw'>
            <CancelRequest closeModal={this.closePopup} tripId={currentTripId} />
          </div>
        </Modal>);
      case 'send_activation_mail':
        return (<div><span className='text-center pb8 iblock wfull'>Didn't receive email</span>
          <button className='btn-filled-pri wfull' onClick={() => this.callbackHandler('Resend_Confirmation')}>Resend
            Confirmation Link
          </button>
        </div>);
      case 'edit_requested_trip':
        return (<Link className='btn-filled-pri wfull at_editPreferencesBtn' to={link.url}>Edit Preferences</Link>);
      case 'show_requested_trip':
      case 'home_path':
        return <Link to={link.url}>{capitalizeEveryFirstLetter(link.label)}</Link>;
      case 'booked_outside':
        return <Link to={link.url} target={'_blank'}>{capitalizeEveryFirstLetter(link.label)}</Link>;
      default:
        return '';
    }
  };

  getParsedMessage = (message) => {
    let finalMessage = message.message;

    if (finalMessage.includes(DUPLICATE_TRIP_MSG)) {
      const messageParts = finalMessage.split('\n');
      finalMessage = messageParts.slice(0, messageParts.length - 1).join('\n');
    }

    message.links.map(link => {
      if (link.type === 'send_activation_mail') {
        const messageParts = finalMessage.split('\n');
        finalMessage = messageParts.slice(0, messageParts.length - 2).join('\n');
      }

      finalMessage = finalMessage.replace(`{{${link.type}}}`, link.label);
    });

    return finalMessage;
  };

  trackSegmentEvents = (event, section, object = '', cta = '') => {
    rescue(() => {
      this.props.trackSegment({
        event,
        section,
        object,
        cta
      });
    });
  };

  getMessageObject = (noQuotesMsg) => {

    const header = noQuotesMsg.header;
    const message = this.getParsedMessage(noQuotesMsg);
    const ctas = [];
    let icon = 'DEFAULT';

    if (message.includes(DUPLICATE_TRIP_MSG)) {
      ctas.push(<a href={`mailto:${TT_EMAIL}`}
                   className='radius2 sfc1 block flexFull flex text-center alignCenter justifyCenter mr5 at_emailUs_CTA'
                   onClick={() => this.trackSegmentEvents(ORGANISM_CLICKED, 'Duplicate lead created section', '', 'Email Us')}>Email
        Us</a>);
      ctas.push(<a href={`tel:${TT_PHONE}`}
                   className='radius2 sfc1 block flexFull flex text-center alignCenter justifyCenter ml5 at_callUs_CTA'
                   onClick={() => this.trackSegmentEvents(ORGANISM_CLICKED, 'Duplicate lead created section', '', 'Call us')}>Call
        Us</a>);
      icon = 'DUPLICATE_TRIP';
    }
    else {
      if (noQuotesMsg.links && noQuotesMsg.links.length) {
        icon = this.getIcon(noQuotesMsg.links[0]) || 'DEFAULT';
        ctas.push(this.getCta(noQuotesMsg.links[0]));
      }
    }

    return {
      header,
      message,
      ctas,
      icon
    };
  };

  render() {
    const { currentTripId, trips } = this.props;
    const currentTrip = trips.byId[currentTripId];
    const message = currentTrip.noQuotesMsg
      ? this.getMessageObject(currentTrip.noQuotesMsg)
      : {
        header: '',
        message: 'Please wait while we are working on your request.',
        ctas: [],
        icon: 'DEFAULT'
      };

    return <RequestProcessingCard message={message} trackSegment={this.trackSegmentEvents} />;
  }
}

LeadSubmittedStage.propTypes = {
  currentTripId: PropTypes.oneOfType([
    PropTypes.number,
    PropTypes.string]).isRequired,
  trips: PropTypes.object,
  activationMailSent: PropTypes.bool,
  activationMailSuccessMsg: PropTypes.string,
  sendActivationMail: PropTypes.func.isRequired,
  reactivateBooking: PropTypes.func.isRequired,
  showRoster: PropTypes.func.isRequired,
  trackSegment: PropTypes.func
};

export default LeadSubmittedStage;
