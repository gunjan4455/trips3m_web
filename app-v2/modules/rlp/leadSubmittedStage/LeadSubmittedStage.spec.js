import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import LeadSubmittedStage from './LeadSubmittedStage';
import RequestProcessingCard from './components/RequestProcessingCard';
import { tripsData } from '../../trip/testData';

configure({ adapter: new Adapter() });

const mockSendActivationMail = jest.fn();
const mockReactivateBooking = jest.fn();

const DEFAULT_PROPS = {
  trips: tripsData,
  currentTripId: 4982978,
  sendActivationMail: mockSendActivationMail,
  reactivateBooking: mockReactivateBooking
};

const ShallowWrapperLeadSubmittedStage = props => {
  return shallow(<LeadSubmittedStage {...props} />);
};

describe('RLP page', () => {
  let props = { ...DEFAULT_PROPS };

  afterEach(() => {
    props = { ...DEFAULT_PROPS };
  });

  it('should render correctly', () => {
    const wrapper = ShallowWrapperLeadSubmittedStage(props);
    expect(wrapper).toMatchSnapshot();
  });

  it('should call closePopup properly', () => {
    const wrapper = ShallowWrapperLeadSubmittedStage(props);
    wrapper.instance().closePopup();
    expect(wrapper.state().closeModal).toEqual(true);
  });

  it('should call callbackHandler properly for reactivate booking', () => {
    const wrapper = ShallowWrapperLeadSubmittedStage(props);
    wrapper.instance().callbackHandler('Reactivate_Booking');
    expect(mockReactivateBooking).toHaveBeenCalledWith({ tripId: props.currentTripId });
  });

  it('should call callbackHandler properly for resend confirmation', () => {
    const wrapper = ShallowWrapperLeadSubmittedStage(props);
    wrapper.instance().callbackHandler('Resend_Confirmation');
    expect(mockSendActivationMail).toHaveBeenCalledWith(props.currentTripId);
  });

  it('should pass proper props in RequestProcessingCard', () => {
    const expected_result = {
      header: '',
      message: 'Please wait while we are working on your request.',
      icon: 'DEFAULT',
      ctas: []
    };
    const wrapper = ShallowWrapperLeadSubmittedStage(props);
    expect(wrapper.find(RequestProcessingCard).prop('message')).toEqual(expected_result);
  });

  it('should call getMessageObject properly', () => {
    const noQuotesMsg = {
      header: 'Request in Progress!',
      message: 'Sit back and relax while our agents handcraft quotes to best fit your requirements.',
      icon: 'DEFAULT',
      links: []
    };

    const expected_result = {
      header: noQuotesMsg.header,
      message: noQuotesMsg.message,
      icon: noQuotesMsg.icon,
      ctas: []
    };

    const wrapper = ShallowWrapperLeadSubmittedStage(props);
    expect(wrapper.instance().getMessageObject(noQuotesMsg)).toEqual(expected_result);
  });

});
