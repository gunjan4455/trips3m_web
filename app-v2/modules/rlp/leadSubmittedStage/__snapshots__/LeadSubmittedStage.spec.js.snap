// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RLP page should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LeadSubmittedStage
    currentTripId={4982978}
    reactivateBooking={[MockFunction]}
    sendActivationMail={[MockFunction]}
    trips={
      Object {
        "allIds": Array [
          "4809460",
          "4982568",
          "4982590",
          "4982978",
        ],
        "byId": Object {
          "4809460": Object {
            "adult": 2,
            "bookingStatusLabel": "Incomplete",
            "budget": "38000..41000",
            "child": 0,
            "currency": "Rupee",
            "exclusivePreferencesValues": Object {
              "cabs": "Included",
              "flights": "Included",
              "hotels": "4 Star",
            },
            "firstDestinationName": "Sri Lanka",
            "fromLoc": "Chennai, Tamil Nadu, India",
            "id": 4809460,
            "info": "",
            "phoneNo": "+919480946000",
            "quotesCount": 3,
            "startDate": "2019-07-04",
            "status": "Cancelled",
            "toLoc": "Sri Lanka",
            "tripDays": 5,
            "tripStage": 3,
          },
          "4982568": Object {
            "adult": 2,
            "bookingStatusLabel": "Incomplete",
            "budget": null,
            "child": 0,
            "currency": null,
            "exclusivePreferencesValues": Object {
              "cabs": "Included",
              "flights": "Included",
              "hotels": "3 Star, 2 Star",
            },
            "firstDestinationName": "Singapore",
            "fromLoc": "Chennai, Tamil Nadu, India",
            "id": 4982568,
            "info": "",
            "phoneNo": "+919498256800",
            "quotesCount": 0,
            "startDate": "2019-08-11",
            "status": "Dormant",
            "toLoc": "Singapore",
            "tripDays": 9,
            "tripStage": 2,
          },
          "4982590": Object {
            "adult": 4,
            "bookingStatusLabel": "Incomplete",
            "budget": "33500..33500",
            "child": 0,
            "currency": "Rupee",
            "exclusivePreferencesValues": Object {
              "cabs": "Included",
              "flights": "Excluded",
              "hotels": "3 Star",
            },
            "firstDestinationName": "Singapore",
            "fromLoc": "Chennai",
            "id": 4982590,
            "info": "",
            "phoneNo": "+919498259000",
            "quotesCount": 0,
            "startDate": "2019-08-19",
            "status": "Dormant",
            "toLoc": "Singapore",
            "tripDays": 7,
            "tripStage": 3,
          },
          "4982978": Object {
            "adult": 4,
            "bookingStatusLabel": "Active",
            "budget": "33500..33500",
            "child": 0,
            "currency": "Rupee",
            "exclusivePreferencesValues": Object {
              "cabs": "Included",
              "flights": "Excluded",
              "hotels": "3 Star",
            },
            "firstDestinationName": "Singapore",
            "fromLoc": "Chennai",
            "id": 4982978,
            "info": "",
            "phoneNo": "+919498297800",
            "quotesCount": 3,
            "startDate": "2019-08-19",
            "status": "Active",
            "toLoc": "Singapore",
            "tripDays": 7,
            "tripStage": 3,
          },
        },
        "loaded": true,
        "loading": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "message": Object {
        "ctas": Array [],
        "header": "",
        "icon": "DEFAULT",
        "message": "Please wait while we are working on your request.",
      },
      "trackSegment": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "message": Object {
          "ctas": Array [],
          "header": "",
          "icon": "DEFAULT",
          "message": "Please wait while we are working on your request.",
        },
        "trackSegment": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
